metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/GrokParser.java

```diff
@@ -132,8 +130,9 @@ public void init() {
 
   @SuppressWarnings("unchecked")
   @Override
-  public List<JSONObject> parse(byte[] rawMessage) {
-    if (grok == null) {
+  public List<JSONObject> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig) {
+    if (grok == null || isGrokPatternUpdated(sensorParserConfig) || isPatternLabelUpdated(sensorParserConfig)) {
+      configure(sensorParserConfig.getParserConfig());
```

Comment:
The timestampField is not what you think it is.  It represents the primary timestamp field of the message that's formatted downstream in the parser, not the last time the pattern was changed.  I would like to see it go away eventually, since you can now do the same thing with Stellar in the transformation phase.

I agree with you though, since it's called for every message that check needs to be as fast as possible (although you wouldn't use a GrokParser for a high velocity sensor anyways).  I think a hash of grokPattern + patternLabel will do the trick.  I will add that in.